src/kudu/tserver/simple_tablet_manager.cc

```diff
@@ -242,6 +243,11 @@ Status TSTabletManager::CreateDistributedConfig(const TabletServerOptions& optio
     }
   }
 
+  if (FLAGS_enable_flexi_raft) {
+    DCHECK(options.topology_config.has_commit_rule());
```

Comment:
nit: we can return bad status instead of dcheck and check, that was server does not crash.
Then we can basically fail turning on raft and stall the bootstrap

Reply:
the fear is if we leave these checks and dchecks around and then the contract changes it will be bad to start crashing servers during rollout.
CHECK and DCHECK should be for things that can be reasoned to be true within your code.
But in some cases, the code will anway crash fail to function because of the violation.
However in raft as a matter of principle, we should keep servers alive and stall in place, 
to allow dbas to debug the issue live, not crash the serve.

Reply:
I had always practiced the opposite on all data infra teams. We would always crash if something meant no progress without manual intervention in order to immediately surface the problem and grab attention of the oncall. Let's discuss this more and evaluate options such as more validations while setting variables etc.?